From efad13b530d1adc209193254d641dc3a7b83816f Mon Sep 17 00:00:00 2001
From: Volker Hilsheimer <volker.hilsheimer@qt.io>
Date: Thu, 22 May 2025 13:13:25 +0200
Subject: [PATCH 1/2] EventCalendar example: balance begin/endResetModel calls

The repopulate() implementation has an early return. Use a scope guard
to make sure that we always balance being and end calls.

Pick-to: 6.9 6.8
Fixes: QTBUG-136947
Change-Id: I8d817ad7bdcae8645afe15479a510c6fd070c88b
Reviewed-by: Dilek Akcay <dilek.akcay@qt.io>
Reviewed-by: Mitch Curtis <mitch.curtis@qt.io>

Upstream-Status: Backport [https://codereview.qt-project.org/c/qt/qtdeclarative/+/647420]

Signed-off-by: Zhang Peng <peng.zhang1.cn@windriver.com>
---
 examples/quickcontrols/eventcalendar/eventmodel.cpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/examples/quickcontrols/eventcalendar/eventmodel.cpp b/examples/quickcontrols/eventcalendar/eventmodel.cpp
index 0b225883ff..22a5ea2386 100644
--- a/examples/quickcontrols/eventcalendar/eventmodel.cpp
+++ b/examples/quickcontrols/eventcalendar/eventmodel.cpp
@@ -2,9 +2,10 @@
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR BSD-3-Clause
 
 #include "eventmodel.h"
-
 #include "sqleventdatabase.h"
 
+#include <QtCore/QScopeGuard>
+
 EventModel::EventModel(QObject *parent)
     : QAbstractListModel(parent)
 {
@@ -76,6 +77,7 @@ bool EventModel::isValid() const
 void EventModel::repopulate()
 {
     beginResetModel();
+    auto endReset = qScopeGuard([this]{ endResetModel(); });
 
     if (!m_eventDatabase || m_date.isNull()) {
         m_events.clear();
@@ -83,6 +85,4 @@ void EventModel::repopulate()
     }
 
     m_events = m_eventDatabase->eventsForDate(m_date);
-
-    endResetModel();
 }
-- 
2.50.0

